webapp/lib/controller.dart

```diff
@@ -216,7 +225,36 @@ void initUI() {
         suggestedReplies.removeWhere((suggestedReply) => updatedIds.contains(suggestedReply.suggestedReplyId));
         suggestedReplies.addAll(updatedReplies);
 
-        _populateReplyPanelView(suggestedReplies);
+        // Update the replies by category map
+        suggestedRepliesByCategory = _groupRepliesIntoCategories(suggestedReplies);
+        // Replace null key (if exists) with empty string
+        if (suggestedRepliesByCategory.containsKey(null)) {
+          suggestedRepliesByCategory[''] = suggestedRepliesByCategory.remove(null);
```

Comment:
@marianamarasoiu  Will there ever be a null and empty string key together?
IOW will this line ever discard/overwrite a category?

Reply:
Good catch, It is possible as category is an optional filed, so when both are present the null category group would overwrite the empty string group. I've now moved the check in the helper file where the grouping gets created, which also fixes this issue. Thank you!